web-common/src/features/dashboards/stores/get-rill-default-explore-state.ts

```diff
@@ -1,17 +1,140 @@
+import {
+  contextColWidthDefaults,
+  LeaderboardContextColumn,
+} from "@rilldata/web-common/features/dashboards/leaderboard-context-column";
+import { getDefaultTimeGrain } from "@rilldata/web-common/features/dashboards/time-controls/time-range-utils";
 import { DEFAULT_TIMEZONES } from "@rilldata/web-common/lib/time/config";
+import { isoDurationToFullTimeRange } from "@rilldata/web-common/lib/time/ranges/iso-ranges";
 import {
   getLocalIANA,
   getUTCIANA,
 } from "@rilldata/web-common/lib/time/timezone";
-import { TimeRangePreset } from "@rilldata/web-common/lib/time/types";
+import {
+  type DashboardTimeControls,
+  TimeRangePreset,
+} from "@rilldata/web-common/lib/time/types";
 import {
   type V1ExploreSpec,
+  type V1MetricsViewSpec,
   V1TimeGrain,
   type V1TimeRangeSummary,
 } from "@rilldata/web-common/runtime-client";
 import { DateTime, IANAZone, Interval } from "luxon";
+import type { MetricsExplorerEntity } from "./metrics-explorer-entity";
+import {
+  DashboardState_ActivePage,
+  DashboardState_LeaderboardSortDirection,
+  DashboardState_LeaderboardSortType,
+} from "@rilldata/web-common/proto/gen/rill/ui/v1/dashboard_pb";
+import { createAndExpression } from "./filter-utils";
+import { TDDChart } from "../time-dimension-details/types";
+
+export function getRillDefaultExploreState(
+  metricsViewSpec: V1MetricsViewSpec,
+  exploreSpec: V1ExploreSpec,
+  timeRangeSummary: V1TimeRangeSummary | undefined,
+) {
+  return <MetricsExplorerEntity>{
+    activePage: DashboardState_ActivePage.DEFAULT,
+
+    whereFilter: createAndExpression([]),
+    dimensionThresholdFilters: [],
+    dimensionsWithInlistFilter: [],
+    dimensionFilterExcludeMode: new Map(),
+    temporaryFilterName: null,
+
+    ...getRillDefaultExploreTimeState(
+      metricsViewSpec,
+      exploreSpec,
+      timeRangeSummary,
+    ),
+    ...getRillDefaultExploreViewState(exploreSpec),
+
+    selectedComparisonDimension: "",
+
+    tdd: {
+      expandedMeasureName: "",
+      chartType: TDDChart.DEFAULT,
+      pinIndex: -1,
+    },
+
+    pivot: {
+      active: false,
+      rows: [],
+      columns: [],
+      sorting: [],
+      expanded: {},
+      columnPage: 1,
+      rowPage: 1,
+      enableComparison: true,
+      activeCell: null,
+      tableMode: "nest",
+    },
```

Comment:
Nit: we'll want these to be their own functions (`getRillDefaultTDDViewState()`, `getRillDefaultPivotView()`) in the future